apis/placement/v1alpha1/stagedupdate_types.go

```diff
@@ -0,0 +1,481 @@
+	// +kubebuilder:default="1h"
+	// +kubebuilder:validation:Pattern="^0|([0-9]+(\\.[0-9]+)?(s|m|h))+$"
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Optional
+	WaitTime metav1.Duration `json:"waitTime,omitempty"`
+}
+
+// StagedUpdateRunStatus defines the observed state of the StagedUpdateRun.
+type StagedUpdateRunStatus struct {
+	// PolicySnapShotIndexUsed records the policy snapshot index of the ClusterResourcePlacement (CRP) that
+	// the update run is based on. The index represents the latest policy snapshot at the start of the update run.
+	// If a newer policy snapshot is detected after the run starts, the staged update run is abandoned.
+	// The scheduler must identify all clusters that meet the current policy before the update run begins.
+	// All clusters involved in the update run are selected from the list of clusters scheduled by the CRP according
+	// to the current policy.
+	// +kubebuilder:validation:Optional
+	PolicySnapShotIndexUsed string `json:"PolicySnapShotIndexUsed"`
+
+	// ApplyStrategy is the apply strategy that the stagedUpdateRun is using.
+	// It is the same as the apply strategy in the CRP when we first start the staged update run.
+	// We will NOT update the apply strategy during the update run even if the apply strategy changes in the CRP.
+	// +kubebuilder:validation:Optional
+	ApplyStrategy v1beta1.ApplyStrategy `json:"appliedStrategy"`
+
+	// StagedUpdateStrategySnapshot is the snapshot of the StagedUpdateStrategy that we are going to use for the update run.
+	// The snapshot is immutable during the update run.
+	// We will apply the strategy to the the list of clusters scheduled by the CRP according to the current policy.
+	// The update run will fail to initialize if the strategy fails to produce a valid list of stages in which each selected
+	// cluster is included in exactly one stage.
+	// +kubebuilder:validation:Optional
+	StagedUpdateStrategySnapshot StagedUpdateStrategySpec `json:"StagedUpdateStrategySnapshot"`
+
+	// Stages list the current status of each stage.
+	// The list is empty if the update run is not started or failed to initialize.
+	// +kubebuilder:validation:Optional
+	StagesStatus []StageStatus `json:"stagesStatus,omitempty"`
+
+	// DeletionStageStatus list the current status of the deletion stage. The deletion stage is
+	// the stage that removes all the resources from the clusters that are not selected by the
+	// current policy after all the update stages are completed.
+	// +kubebuilder:validation:Optional
+	DeletionStageStatus StageStatus `json:"deletionStageStatus,omitempty"`
+
+	// CurrentUpdatingStage is the name of the stage that is in the middle of the update.
+	// If the update run is not started, the value is an empty string.
+	// If we are in the DeletionStage, the name is called "deletionStage".
+	// +kubebuilder:validation:Optional
+	CurrentUpdatingStage string `json:"currentUpdatingStage,omitempty"`
```

Comment:
Can we remove this as this information is already part of stagesstatus?

What about current cluster (that's in stagesstatus too)?